### PR TITLE
[core] Manage the lookup file disk cache in task granularity

### LINKED ISSUE #3848: [Feature] Manage the lookup file disk cache in task granularity
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Motivation

The LookupLevels create a cache for each bucket in the partition. Therefore, if a task has multiple buckets/partitions to write to, the total disk usage can become almost unlimited. This could lead to exceeding disk quotas errors in Kubernetes environments. As a solution, I propose managing the disk cache at the task level, similar to how CacheManager operates.

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by Aitozi**
### Purpose

Linked issue: close #3848

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/mergetree/LookupLevels.java [hunk 1]
```diff
@@ -207,12 +211,8 @@ private LookupFile createLookupFile(DataFileMeta file) throws IOException {
         return new LookupFile(localFile, file, lookupStoreFactory.createReader(localFile, context));
     }
 
-    @Override
-    public void close() throws IOException {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/mergetree/LookupLevels.java:211]: Should we clean all files when closing?

Comment (JingsongLi) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/LookupLevels.java:211]: But by default, there is no max size configuration.

Comment (Aitozi) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/LookupLevels.java:211]: Evict by the retention time?

Comment (Aitozi) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/LookupLevels.java:211]: Before the LookupLevels's cache file will be cleared when the writer is idled (no data written to in the latest snapshot). 

Now, the cache file will still be there after the idle writer closed and evict by the retention time by default.

Comment (JingsongLi) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/LookupLevels.java:211]: Yes, it is better to remove all files when closing this writer.

Comment (JingsongLi) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/LookupLevels.java:211]: You can manage a collection in `LookupLevels`.

Comment (Aitozi) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/LookupLevels.java:211]: > You can manage a collection in `LookupLevels`.

Thanks, Make sense to me.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**